google-cloud-bigtable/test/google/cloud/bigtable/conformance_test.rb

```diff
@@ -0,0 +1,90 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "helper.rb"
+require_relative "../../../../conformance/v2/proto/google/cloud/conformance/bigtable/v2/tests_pb.rb"
+
+##
+# This suite of unit tests is dynamically generated from the contents of
+# `conformance/v2/*.json`, using the protobuf types defined in
+# `conformance/v2/proto/google/cloud/conformance/bigtable/v2/tests_pb.rb`, which
+# was manually generated from
+# `conformance/v2/proto/google/cloud/conformance/bigtable/v2/tests.proto`.
+#
+# The `conformance/v2` directory was manually imported (copied) from
+# https://github.com/googleapis/conformance-tests/tree/master/bigtable/v2.
+#
+# See [Protocol Buffers - Ruby Generated
+# Code](https://developers.google.com/protocol-buffers/docs/reference/ruby-generated)
+# for instructions in case `tests.proto` is updated.
+#
+class ReadRowsTest < MockBigtable
+  let(:instance_id) { "test-instance" }
+  let(:table_id) { "test-table" }
+
+  def self.build_test_for test, index
+    define_method("test_#{index}: #{test.description}") do
+      # start: test method body
+      mock = Minitest::Mock.new
+      bigtable.service.mocked_client = mock
+
+      table = bigtable.table(instance_id, table_id)
+
+      resp = Google::Bigtable::V2::ReadRowsResponse.new(chunks: test.chunks.to_a)
+
+      mock.expect :read_rows, [resp], [
+                              table_path(instance_id, table_id),
+                              rows: Google::Bigtable::V2::RowSet.new,
+                              filter: nil,
+                              rows_limit: nil,
+                              app_profile_id: nil
+                            ]
+
+      results = test.results.to_a
+      if results.map(&:error).include? true
+        expect do
+          table.read_rows.to_a # to_a is needed because Enum lazily invokes read_rows
+        end.must_raise Google::Cloud::Bigtable::InvalidRowStateError
```

Comment:
It looks like this doesn't test the results before the error one. The test should be able to get the expected results before it errors.
I could just be misreading it, of course...

Reply:
Absolutely right. I'll work on an update.